pkg/operator/operator/validations.go

```diff
@@ -744,3 +789,54 @@ func getValidationK8sResources() ([]kunstructured.Unstructured, *kresource.Quant
 
 	return virtualServices, maxMem, err
 }
+
+func validateDockerImagePath(image string) error {
+	if _, err := cr.ValidateImageVersion(image); err != nil {
+		return err
+	}
+
+	dockerAuth := dockerlib.NoAuth
+	if regex.IsValidECRURL(image) {
+		operatorID, _, err := config.AWS.GetCachedAccountID()
+		if err != nil {
+			return err
+		}
+		registryID := aws.GetAccountIDFromECRURL(image)
+
+		if strings.Compare(operatorID, registryID) != 0 {
+			return ErrorRegistryAccountIDMismatch(registryID, operatorID)
+		}
+
+		ecrAuthConfig, err := config.AWS.GetECRAuthConfig()
+		if err != nil {
+			return err
+		}
+
+		dAuth, err := dockerlib.EncodeAuthConfig(dockertypes.AuthConfig{
```

Comment:
This is totally up to you, it's more of a style/clarity consideration: since `err` is already defined, you could do:

```go
dockerAuth, err = dockerlib.EncodeAuthConfig(dockertypes.AuthConfig{
```

and remove `dockerAuth = dAuth`